### PR TITLE
Added new methods for specifying other resource limits/requests

### PR DESCRIPTION
**Pull request by foxish**
Added some additional config options that can be overridden from a form.
Tested against a form in https://gist.github.com/foxish/3407150729b106a05811d672d905c27b

@yuvipanda @liyinan926 PTAL

### CODE REVIEW

File: kubespawner/spawner.py [hunk 1]
```diff
@@ -771,6 +795,24 @@ def get_pod_manifest(self):
             extra_containers=self.singleuser_extra_containers
         )
 
+    def get_cpu_limit(self):
```
Comment (minrk) [inline kubespawner/spawner.py]: Why all these getter methods? One of the nice things about Python is that you don't need getter methods to get all of the benefits of getters. Subclasses can define properties or extend the traitlets here without needing to add any methods on the base class.

Comment (foxish) [reply kubespawner/spawner.py]: I was under the impression that we need these to be present to facilitate overriding as in https://gist.github.com/foxish/3407150729b106a05811d672d905c27b. If we can performing overriding of that kind without really adding those getters, that would be ideal. Can you elaborate a bit more on the mechanism by which one can do that?

Comment (yuvipanda) [reply kubespawner/spawner.py]: @foxish you can do so by overriding start like https://github.com/berkeley-dsep-infra/datahub/blob/staging/datahub/config.yaml#L12

**Comment by yuvipanda** [comment]
@foxish thank you for the patch! I believe this fixes #93 too! Can you address @minrk's comments and then we can merge this?

**Comment by minrk** [comment]
👍 to merge without the getters. You can also use a property:

```python
class MySpawner(KubeSpawner):
    @property
    def mem_limit(self):
        return construct_mem_limit(self.user)
```

which is the same as having a getter, it just doesn't need to be defined as a function on the parent class for the child to have the extra logic.

**Comment by foxish** [comment]
@minrk @yuvipanda PTAL, removed the getters. Thanks for the review.

**Comment by yuvipanda** [comment]
Can you also add a test for this, @foxish? It's in test_objects.py I believe.

File: tests/test_objects.py [hunk 2]
```diff
@@ -631,6 +631,121 @@ def test_make_pod_with_extra_containers():
         "apiVersion": "v1"
     }
 
+def test_make_pod_with_extra_containers():
+    """
+    Test specification of a pod with initContainers
```
Comment (yuvipanda) [inline tests/test_objects.py]: this description is wrong I think?

**Comment by yuvipanda** [comment]
This lgtm, I'll merge once the tests pass!

**Pull request merged**